<launch>
    <!-- Start the GMapping SLAM Node -->
    <node pkg="gmapping" type="slam_gmapping" name="gmapping" respawn="true">
	<param name="sigma" value="0.05"/>
	<param name="kernelSize" value="1"/>
	<param name="lstep" value="0.15"/>
	<param name="astep" value="0.15"/>
	<param name="iterations" value="10"/>
	<param name="lsigma" value="0.075"/>
	<param name="ogain" value="10.0"/>
	<param name="lskip" value="5"/>
	<param name="srr" value="0.1"/>
	<param name="srt" value="0.1"/>
	<param name="str" value="0.1"/>
	<param name="stt" value="0.1"/>
	<param name="linearUpdate" value="0.5"/>
	<param name="angularUpdate" value="0.1"/>
	<param name="temporalUpdate" value="0.10"/>
	<param name="throttle_scans" value="2"/>
        
        <param name="maxUrange" value="30.0" /> <!-- The maximum usable range of the laser. A beam is cropped to this value. (meters) -->
        <param name="particles" value="3" /> <!-- Number of particles in the filter -->
        <param name="delta" value="0.05" /> <!-- Processing parameters (resolution of the map) -->
        <param name="xmin" value="-30.0" /> <!-- Initial map size -->
        <param name="xmax" value="30.0" /> <!-- Initial map size -->
        <param name="ymin" value="-30.0" /> <!-- Initial map size -->
        <param name="ymax" value="30.0" /> <!-- Initial map size -->
        <param name="angularUpdate" value="0.1" /> <!-- Process a scan each time the robot rotates this far (radians) -->
        <param name="linearUpdate" value="0.1" /> <!-- Process a scan each time the robot translates this far (meters) -->
        <param name="map_update_interval" value="1.0" />
        <!-- 
        How long (in seconds) between updates to the map.
        Lowering this number updates the occupancy grid more often,
        at the expense of greater computational load.
        -->
        <param name="resampleThreshold" value="0.3" /> <!-- The neff based resampling threshold (?) -->
        <param name="llsamplerange" value ="0.05" /> <!-- Translational sampling range for the likelihood -->
        <param name="llsamplestep" value ="0.05" /> <!-- Translational sampling range for the likelihood -->
        <param name="lasamplerange" value ="0.5" /> <!-- Angular sampling range for the likelihood -->
        <param name="lasamplestep" value ="0.5" /> <!-- Angular sampling range for the likelihood -->

    	<!-- param name="odom_frame" value="/cata_odom_combined_frame" / --> # After outputting an Odometry message since robot_pose_ekf does NOT
	<param name="odom_frame" value="/odom_combined" />
        <param name="base_frame" value="/base_link" />
  </node>
   
</launch>
